metrics-proxy/src/main/java/ai/vespa/metricsproxy/service/NodeMetricGatherer.java

```diff
@@ -0,0 +1,210 @@
+package ai.vespa.metricsproxy.service;
+
+import ai.vespa.metricsproxy.core.MetricsManager;
+import ai.vespa.metricsproxy.metric.dimensions.ApplicationDimensions;
+import ai.vespa.metricsproxy.metric.dimensions.NodeDimensions;
+import ai.vespa.metricsproxy.metric.model.MetricsPacket;
+import ai.vespa.metricsproxy.metric.model.StatusCode;
+import ai.vespa.metricsproxy.metric.model.json.YamasJsonUtil;
+import com.google.inject.Inject;
+import com.yahoo.vespa.defaults.Defaults;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Fetches miscellaneous system metrics for node, including
+ *  - Current coredump processing
+ *  - Health of Vespa services
+ *  - Host life
+ *
+ * @author olaa
+ */
+
+
+public class NodeMetricGatherer {
+
+    private static final int COREDUMP_AGE_IN_MINUTES = 12600;
+    private static final JSONObject ROUTING_JSON = createRoutingJSON();
+    private final VespaServices vespaServices;
+    private final ApplicationDimensions applicationDimensions;
+    private final NodeDimensions nodeDimensions;
+    private final MetricsManager metricsManager;
+    private final FileWrapper fileWrapper;
+
+    private static final Logger logger = Logger.getLogger(NodeMetricGatherer.class.getSimpleName());
+
+    @Inject
+    public NodeMetricGatherer(MetricsManager metricsManager, VespaServices vespaServices, ApplicationDimensions applicationDimensions, NodeDimensions nodeDimensions) {
+        this(metricsManager, vespaServices, applicationDimensions, nodeDimensions, new FileWrapper());
+    }
+
+    public NodeMetricGatherer(MetricsManager metricsManager,
+                              VespaServices vespaServices,
+                              ApplicationDimensions applicationDimensions,
+                              NodeDimensions nodeDimensions,
+                              FileWrapper fileWrapper) {
+        this.metricsManager = metricsManager;
+        this.vespaServices = vespaServices;
+        this.applicationDimensions = applicationDimensions;
+        this.nodeDimensions = nodeDimensions;
+        this.fileWrapper = fileWrapper;
+    }
+
+    public List<MetricsPacket> gatherMetrics()  {
+        List<MetricsPacket.Builder> metricPacketBuilders = new ArrayList<>();
+        metricPacketBuilders.addAll(coredumpMetrics());
+        metricPacketBuilders.addAll(serviceHealthMetrics());
+        metricPacketBuilders.addAll(hostLifeMetrics());
+
+        List<MetricsPacket> metricPackets = metricPacketBuilders.stream().map(metricPacketBuilder -> {
```

Comment:
I think you can simplify this by just using the dot operator and skip the temporary `metricsPackets` variable.